### DISCUSSION TITLE
Independent versioning for each sub-project

### DISCUSSION THREAD

**Discussion by danielo-unity3d** [GitTools/GitVersion#4029] [Q&A] [answered]
Hi, 
Has anyone here ever felt the need to treat the managed sources (repository contents) as multiple, independent modules and sub-project, with version numbers incrementing independently?
This might imply that we are no longer looking at branch names and tags (metadata) to tell us what the current version number is, and that we would like the basis for the versioning to be embedded in each of the `VersionInfo` (or `csproj`) files of the corresponding projects. 
This follows from the observation that git itself can return only the subset of those commits that impact the target subset of the files that constitute the affected module. In that sense, we are treating the subset as an independent repository.
Either the versions are committed to the files, and are incremented manually by each dev as they submit PRs (possibly causing problems as they merge), or they are managed through GitHub actions or some similar mechanism.
We have two major projects in the same repository (they share some subset of the sources), and they are intertwined in such a way as to make breaking them up problematic (also includes test suites that depend on both). We would like for each project to have an independent version (which undoubtedly reflects the amalgamation of the versions of the dependent modules).

Anybody?
Thanks in advance.

**Comment by HHobeck** [comment, marked as answer]
Please take a look to this issue:
- https://github.com/GitTools/GitVersion/issues/2436
